payments-ui-core/src/main/java/com/stripe/android/ui/core/elements/HtmlText.kt

```diff
@@ -0,0 +1,212 @@
+package com.stripe.android.ui.core.elements
+
+import android.annotation.SuppressLint
+import android.content.Intent
+import android.graphics.Typeface
+import android.net.Uri
+import android.text.style.ForegroundColorSpan
+import android.text.style.ImageSpan
+import android.text.style.StyleSpan
+import android.text.style.URLSpan
+import android.text.style.UnderlineSpan
+import androidx.compose.foundation.Image
+import androidx.compose.foundation.gestures.detectTapGestures
+import androidx.compose.foundation.layout.fillMaxHeight
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.text.BasicText
+import androidx.compose.foundation.text.InlineTextContent
+import androidx.compose.foundation.text.appendInlineContent
+import androidx.compose.material.MaterialTheme
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.input.pointer.pointerInput
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.semantics.semantics
+import androidx.compose.ui.text.AnnotatedString
+import androidx.compose.ui.text.Placeholder
+import androidx.compose.ui.text.PlaceholderVerticalAlign
+import androidx.compose.ui.text.SpanStyle
+import androidx.compose.ui.text.TextLayoutResult
+import androidx.compose.ui.text.TextStyle
+import androidx.compose.ui.text.buildAnnotatedString
+import androidx.compose.ui.text.font.FontStyle
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.text.style.TextDecoration
+import androidx.compose.ui.text.style.TextOverflow
+import androidx.compose.ui.unit.dp
+import androidx.core.text.HtmlCompat
+import com.stripe.android.ui.core.PaymentsTheme
+
+private const val LINK_TAG = "URL"
+
+@Composable
+internal fun HtmlText(
+    html: String,
+    imageGetter: Map<String, Int>,
+    modifier: Modifier = Modifier
+) {
+    val inlineContentMap = imageGetter.entries.associate { (key, value) ->
+        val painter = painterResource(value)
+        val height = painter.intrinsicSize.height
+        val width = painter.intrinsicSize.width
+        val newWidth = MaterialTheme.typography.body1.fontSize * (width / height)
```

Comment:
I'll need to update this to follow the rest of the typography stuff. I'll make a note.